articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,147 @@
+
+## Step 4 - Configure NGINX at 80 for HTTP and 443 for HTTPS
+Now that we have our apps running and our DNS records ready. We can start configuring our NGINX to make it all work.
+Navigate into the Nginx folder and locate  ```nginx.conf``` .
+
+Add these configurations inside the HTTP block.
+
+### Step 4.1 - HTTP
+
+```
+server {       
+     listen       80;
+     server_name  server.domain;
+     location / {
+          proxy_pass "http://localhost:3000" ;
+     }
+}
+server {       
+     listen       80;
+     server_name  client.domain;
+
+     location / {
+          proxy_pass "http://localhost:3001" ;
+     }
+}
+```
+
+### Step 4.2 - HTTPS
+
+```
+server{
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name server.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3000/";
+    }
+
+    ssl_certificate <location of SSL certificate>
+    ssl_certificate_key <location of SSL certificate Key>
+}
+server{
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name client.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3001/";
+    }
+
+    ssl_certificate <location of SSL certificate>
+    ssl_certificate_key <location of SSL certificate Key>
+}
+```
+
+**NOTE** 
+These are the basic minimum configurations required to successfully configure NGINX for reverse proxying. Feel free to explore other config parameters as well. 
+
+Change the domain name to your domain.
+For SSL Certificate and Key, you can get them from your SSL provider. If you don't have one, Use this free service [LetsEncrypt](https://letsencrypt.org/). Follow their documentation to get Free SSL instantly!
+
+
+
+## Step 5 - Save and Restart
+
+After editing, save your changes. Use ```nginx -t``` command to test your changes before actually reloading NGINX. It is good practice do that to make sure your server doesn't crash if there were any errors in your config file. Once you get a message that the test is successful, you can go ahead and restart your NGINX.
+Open the browser and enter the URLs to find your applications running on the corresponding URLs you configured.
+
+**Important Note**
+> It is important to note that the clients see your applications as a whole where the face of it being NGINX running at port 80 or 443 according to the protocol used. Once the request reaches NGINX, it is being internally routed to the corresponding application, There is no need to open all of your PORTS, in this case, 3000 & 3001 to the internet. Just enabling 80 and 443 would be sufficient as NGINX routes the traffic internally. This is a highly secure architecture as each application need not be taken care individually rather the single entry point i.e NGINX is the only place which needs to be focussed.
```

Comment:
Please revise this paragraph. Some parts of this are confusing and don't make sense such as "It is important to note that the clients see your applications as a whole where the face of it being NGINX running at port 80 or 443 according to the protocol used." 

and "s each application need not be taken care individually rather the single entry point i.e NGINX is the only place which needs to be focussed."

Reply:
How about this?
"  It is important to note that,  the clients see all your applications as a black box. With NGINX being the Gateway at Port 80 or 443, depending on the Protocol followed."

"This is a  secure architecture. Because all the traffic passes through the Secure NGINX server then gets redirected."

Reply:
Much better, @HarishTeens. Thanks. You could rephrase a bit more for increased clarity though. How about the following? 

"Using NGINX secures your server because it routes the traffic internally. Instead of having to open up all of your ports, in this case 3000 and 3001, to the internet, just 80 and 443 will do the trick. 

This is because all traffic passes through the secure NGINX server (like a gateway) and is redirected to the correct application. Using a reverse proxy like NGINX is much more secure that opening up several ports for every application you deploy because of the increased risk a hacker will use an open port for malicious activity."

Reply:
This is very much clear, thank you so much @louisefindlay23 .